app/models/katello/host/subscription_facet.rb

```diff
@@ -35,6 +36,19 @@ def import_database_attributes(consumer_params)
         end
       end
 
+      def update_consumer_type(consumer_params)
+        if consumer_params.try(:[], 'type').try(:[], 'label') == 'hypervisor'
+          self.consumer_type = 'hypervisor'
+        elsif host.facts.try(:[], 'virt.is_guest') == "true"
```

Comment:
@thomasmckay @jlsherrill  I switched this to pull from the `host.facts` as they appear to be get updated before this would be called and I guess should be the canonical source of information for this field.